library/Zend/Db/Statement.php

```diff
@@ -134,14 +134,17 @@ protected function _prepare($sql)
      */
     protected function _parseParameters($sql)
     {
-        $sql = $this->_stripQuoted($sql);
-
-        // split into text and params
-        $this->_sqlSplit = preg_split('/(\?|\:[a-zA-Z0-9_]+)/',
-            $sql, -1, PREG_SPLIT_DELIM_CAPTURE|PREG_SPLIT_NO_EMPTY);
+        if ($sql !== null) {
```

Comment:
How will the following `foreach` be executed if `$sql === null`?

Reply:
fixed